internal/services/gateway/action/user.go

```diff
@@ -997,3 +997,33 @@ func (h *ActionHandler) UserCreateRun(ctx context.Context, req *UserCreateRunReq
 
 	return h.CreateRuns(ctx, creq)
 }
+
+func (h *ActionHandler) GetUserGitSource(ctx context.Context, remoteSourceRef, userID string) (gitsource.GitSource, *cstypes.RemoteSource, *cstypes.LinkedAccount, error) {
+	rs, _, err := h.configstoreClient.GetRemoteSource(ctx, remoteSourceRef)
+	if err != nil {
+		return nil, nil, nil, util.NewAPIError(util.KindFromRemoteError(err), errors.Wrapf(err, "failed to get remote source %q", remoteSourceRef))
+	}
+
+	linkedAccounts, _, err := h.configstoreClient.GetUserLinkedAccounts(ctx, userID)
+	if err != nil {
+		return nil, nil, nil, util.NewAPIError(util.KindFromRemoteError(err), errors.Wrapf(err, "failed to get linked accounts for user %q", userID))
```

Comment:
Other errors for such remote call are `"failed to get user %q linked accounts"`.